### ISSUE TITLE
Add CTR metric

### ISSUE THREAD

**Issue by dbish** [pytorch-labs/torcheval#27]
### 🚀 The feature

**Objective**   
We want to implement a functional metric and class metric to calculate CTR (click through rate) based on inputs
CTR = sum( I(click) * weight) / sum(weights)

Examples:   

input: torch.tensor([0, 1, 0, 1, 1, 0, 0, 1])
weight: 1.0 (default value)
ctr: torch.tensor(0.5)

input: torch.tensor([0, 1, 0, 1, 1, 0, 0, 1])
weight: torch.tensor([1.0, 2.0, 1.0, 2.0, 1.0, 2.0, 1.0, 2.0])
ctr: torch.tensor(0.58333)

For multi-tasks situation
input: torch.tensor([[0, 1, 0, 1], [1, 0, 0, 1]])
weight: 1.0 (default value)
ctr: torch.tensor([0.5, 0.05])

### Motivation, pitch

Click-through rates are important for evaluating model performance and a commonly expected out-of-the-box metric someone might want to get started with.

**Comment by ananthsub** [comment]
Do you think the `Mean` metric we have here could be used for this? https://github.com/pytorch-labs/torcheval/blob/8532596ffa6829819ebdc757ab25144ddda0635f/torcheval/metrics/aggregation/mean.py#L61-L84

we compute a weighted average. however it does not currently support the multi-task usage

**Comment by dbish** [comment]
@ninginthecloud sent the request/requirements so I'll let her comment on the use of Mean vs. a new CTR metric. i'm just the implementer getting to know the library with a new metric :)

**Comment by DuYicong515** [comment]
I feel `Mean` metric is a more generalized version. 
The thing we can get out of a separate CTR metric is we can check the input dimensions and an easy discovery for ranking usages. I'm okay with either adding multi task support to existing Mean metric or adding a separate CTR metric. 

Leave the decision to @ninginthecloud

**Comment by ninginthecloud** [comment]
Hi, @dbish and @DuYicong515 , Let's keep a separate CTR metric implementation, although mathematically they are similar, it will be easy for users to discover the metrics. 
In summary, let's write a CTR metric with multi-task support (see example https://github.com/pytorch-labs/torcheval/blob/main/torcheval/metrics/functional/classification/binary_normalized_entropy.py)

**Comment by dbish** [comment]
Closing as won't fix for now (moving back to a an open ramp up task as I won't have time now).
